pkg/bgpv1/manager/reconcile_test.go

```diff
@@ -209,6 +222,19 @@ func TestNeighborReconciler(t *testing.T) {
 			},
 			err: nil,
 		},
+		{
+			name: "unsupported neighbor update",
+			neighbors: []v2alpha1api.CiliumBGPNeighbor{
+				{PeerASN: 64124, PeerAddress: "192.168.0.1/32"},
+				{PeerASN: 64124, PeerAddress: "192.168.0.2/32"},
+			},
+			newNeighbors: []v2alpha1api.CiliumBGPNeighbor{
+				{PeerASN: 64124, PeerAddress: "192.168.0.1/32", PeerPort: v2alpha1api.ToPeerPort(uint16(42424))},
+				{PeerASN: 64124, PeerAddress: "192.168.0.2/32"},
+			},
+			err: errors.New(`failed while reconciling neighbor 192.168.0.1/32 42424 64124:
+			failed while adding peer 192.168.0.1/32 64124: can't overwrite the existing peer: 192.168.0.1`),
+		},
```

Comment:
Why we can't change the port?

Reply:
For simplicity, this PR follows the `PeerAddress` implementation. I created #25869 to track this requirement.

Reply:
I was following the PeerAddress implementation. I'll update the PR to allow updating the PeerPort.